### PR TITLE
UI: add guidance handling, show toggle only on relevant tab

### PR DESCRIPTION
**Pull request by sofa-tata**

### CODE REVIEW

File: lib/components/Wizard/RightSidebar/RightSidebar.tsx [hunk 1]
```diff
@@ -51,11 +57,34 @@ function RightSidebar({
...
-  const initialFormattedTabs: FormattedTab[] = tabs.map((tab) => {
+  useEffect(() => {
+    const foundSection = guidance.find(
+      (sectionGuidance) => sectionGuidance.section === selectTab
+    )
+    if (foundSection) {
+      setCurrentGuidancePart(foundSection)
+    } else {
+      setCurrentGuidancePart(null)
+    }
```
Comment (mvoloshin0458) [inline lib/components/Wizard/RightSidebar/RightSidebar.tsx:84]:
```suggestion
   setCurrentGuidancePart(foundSection ?? null)
```

File: lib/components/Wizard/RightSidebar/RightSidebar.tsx [hunk 2]
```diff
@@ -67,6 +96,23 @@ function RightSidebar({
...
+    key: 'configGuidance',
+    title: guidanceTabTitle || 'Configuration Guidance',
+    isGuidanceTab: true,
+    shouldShow:
+      guidance.length > 0 &&
+      guidance.some(
+        ({ sectionDescription, inputsDescription }) =>
+          !!sectionDescription ||
+          inputsDescription?.some(({ description }) => !!description)
+      )
```
Comment (mvoloshin0458) [inline lib/components/Wizard/RightSidebar/RightSidebar.tsx:109]: the value you return in a `some` callback is automatically converted to a boolean, so `!!` can be omitted

File: lib/components/Wizard/RightSidebar/RightSidebar.tsx [hunk 3]
```diff
@@ -131,10 +190,15 @@ function RightSidebar({
...
+      if (!hasRenderedWithValue && guidanceTab) {
+        setActiveTab(guidanceTab)
+        setHasRenderedWithValue(true)
+      } else {
+        setActiveTab(incompleteTab)
+      }
     }
     setFormattedTabs(updatedTabs)
-  }, [jsonValue])
+  }, [JSON.stringify(jsonValue)])
```
Comment (mvoloshin0458) [inline lib/components/Wizard/RightSidebar/RightSidebar.tsx:201]: can updating of `jsonValue` be optimized to omit stringifying it each render?

File: lib/components/Wizard/RightSidebar/RightSidebar.tsx [hunk 4]
```diff
@@ -179,28 +243,36 @@ function RightSidebar({
...
+    if (activeTab?.isParsableTab) {
+      return (
+        <JsonSidebar
+          allowCopy={true}
+          jsonValue={
+            activeTab?.content ||
+            (isParserTab(activeTab) &&
+              jsonValue &&
+              activeTab?.parser(jsonValue)) ||
+            'Invalid Value. Please provide the `incompleteTab` parameter in order to list all the incomplete sections.'
```
Comment (mvoloshin0458) [inline lib/components/Wizard/RightSidebar/RightSidebar.tsx:259]: Is this issue for users or developers of the config?

Comment (sofa-tata) [reply lib/components/Wizard/RightSidebar/RightSidebar.tsx:259]: For the ones who uses the Wizard package (like the `terraform-wizard` for now)

File: lib/components/Wizard/RightSidebar/RightSidebar.tsx (hunk 2 above)
Comment (sofa-tata) [reply lib/components/Wizard/RightSidebar/RightSidebar.tsx:109]: 👍 thanks

File: lib/components/Wizard/RightSidebar/RightSidebar.tsx (hunk 3 above)
Comment (sofa-tata) [reply lib/components/Wizard/RightSidebar/RightSidebar.tsx]: realized I don't really need to stringify here, so removed

**Review by gonenidan — APPROVED** [review]

**Review by mvoloshin0458 — APPROVED** [review]

**Pull request merged**